examples/anon_lua/module.c

```diff
@@ -0,0 +1,134 @@
+/* 
+ * Copyright (c) 2024, The Pallene Developers
+ * Pallene Tracer is licensed under the MIT license.
+ * Please refer to the LICENSE and AUTHORS files for details
+ * SPDX-License-Identifier: MIT 
+ */
+
+/* Static use of the library would suffice. */
+#define PT_IMPLEMENTATION
+#include <ptracer.h>
+
+/* ---------------- FOR LUA INTERFACE FUNCTIONS ---------------- */
+#define PREPARE_FINALIZER()                             \
+    int _base = lua_gettop(L);                          \
+    lua_pushvalue(L, lua_upvalueindex(2));              \
+    lua_toclose(L, -1); /* The finalizer fn will run whenever out of scope. */
+#define MODULE_LUA_FRAMEENTER(sig)                      \
+    pt_cont_t *cont = (pt_cont_t *)                     \
+        lua_touserdata(L, lua_upvalueindex(1));         \
```

Comment:
It's good practice. In this way someone reading the code will be able to know the function we are calling most likely returning `void *` even tho the reader may not aware of the function signature.

As far as I can remember, in C++ you don't necessarily have to do casting for pointers (correct me if I am wrong, I haven't been fiddling with C++ for while). Tho you have some robust way to do cast in C++ like static_cast, reinterpret_cast etc.

Reply:
I'm not convinced by that best-practice argument...

- If the function were to return a different pointer type instead of void, that type cast will silently convert it.
- We don't cast the return value of other functions that return pointers, why should we do that for void pointers?

The C++ issue is that C++ does not implicitly convert void* to other pointer types. In C we do not have to cast the return value of malloc, but in C++ we do.

Reply:
Ohh yeah, another thing. The header may also be used in a C++ environment :). We are not bound to Pallene anymore.

About the best-practice: 
1. I have seen these pointer casting many times in C that I can count. Heck, I have also seen great C code where the `void *` pointer returned by malloc also get casted :smile:. I have also seen stuff like this:
```C
(void) fprintf(stderr, "Some error");
```
written by great C programmers, which is completely useless even by my standards. But people do it anyway, because It gives you clarity.
2. It's just a good coding practice (Not best-practice that you must follow to be a great programmer).

Reply:
The (void) cast to ignore the return value is a totally different thing.

I can tolerate the (void*) casts if we commit to supporting C++ but in that case we should probably have actual tests for it... (I still disagree that it's "best practice")

Reply:
We would rather not put C++ tests for now.

I agree it's a *GOOD* practice *NOT BEST* practice as I have stated earlier. It may look weird to your eyes but it's not a huge deal to be honest that we should waste time discussing.